include/token.h

```diff
@@ -0,0 +1,31 @@
+#ifndef TOKEN_H
+#define TOKEN_H
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+struct token;
```

Comment:
Why hide the struct's declaration in the source file? We don't do it for the other structs, why `token`?